src/DataStructures/VectorImpl.hpp

```diff
@@ -530,3 +530,83 @@ std::ostream& operator<<(std::ostream& os,
   MAKE_MATH_ASSIGN_EXPRESSION_POINTERVECTOR(-=, VECTOR_TYPE) \
   MAKE_MATH_ASSIGN_EXPRESSION_POINTERVECTOR(*=, VECTOR_TYPE) \
   MAKE_MATH_ASSIGN_EXPRESSION_POINTERVECTOR(/=, VECTOR_TYPE)
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief Defines the MakeWithValueImpl apply specialization
+ *
+ * \details The `MakeWithValueImpl<VECTOR_TYPE, VECTOR_TYPE>::apply(VECTOR_TYPE,
+ * VECTOR_TYPE::value_type)` specialization defined by this macro produces an
+ * object with the same size as the type argument, initialized with the value
```

Comment:
One more minor fix since you are still working on this:  type argument -> input argument  and I would put `input` and `value` in ticks.